tezos/encoding/src/binary_reader.rs

```diff
@@ -60,12 +60,11 @@ macro_rules! safe {
 
 }
 
-pub struct BinaryReader {}
+pub struct BinaryReader;
 
 impl BinaryReader {
-
     pub fn new() -> Self {
-        BinaryReader {}
+        Self
```

Comment:
Its common, while defininf empty struct to leave no body (`pub struct BinaryReader;`), instead of empty body (`pub struct BinaryReader {}`), to better represent your intentions. But, then calling new structure initialization with empty body become invalid